platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/ConfigurationAdminExt.java

```diff
@@ -670,6 +682,15 @@ public Map getFactoryPidObjectClasses() {
         return metatypeList;
     }
 
+    public static boolean isPermittedToViewService(String servicePid) {
+        KeyValueCollectionPermission serviceToCheck = new KeyValueCollectionPermission(
+                "view-service.pid",
+                new KeyValuePermission("service.pid", Sets.newHashSet(servicePid)));
+        Subject subject = ThreadContext.getSubject();
```

Comment:
I think you should be using SecurityUtils.getSubject() here.  Is there a reason you had to do it this way?